### PR TITLE
optimize: optimize metrics rt statistics

### PR DESCRIPTION
**Pull request by slievrly**
Signed-off-by: slievrly <slievrly@163.com>

- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did

optimize: optimize metrics

### CODE REVIEW

File: server/src/main/java/io/seata/server/coordinator/DefaultCore.java [hunk 1]
```diff
@@ -343,10 +347,13 @@ public boolean doGlobalRollback(GlobalSession globalSession, boolean retrying) t
                 return result;
             }
         }
+        if (!retrying) {
```
Comment (a364176773) [inline server/src/main/java/io/seata/server/coordinator/DefaultCore.java:350]: if(success){
if(retrying){}else{}
}

File: server/src/main/java/io/seata/server/coordinator/DefaultCore.java [hunk 2]
```diff
@@ -254,13 +256,15 @@ public boolean doGlobalCommit(GlobalSession globalSession, boolean retrying) thr
...
             }
             if (!retrying) {
+                //contains tcc branch
                 globalSession.setStatus(GlobalStatus.Committed);
+                MetricsPublisher.postSessionDoneEvent(globalSession, false);
             }
         }
         // if it succeeds and there is no branch, retrying=true is the asynchronous state when retrying. EndCommitted is
         // executed to improve concurrency performance, and the global transaction ends..
         if (success && globalSession.getBranchSessions().isEmpty() && retrying) {
```
Comment (a364176773) [inline server/src/main/java/io/seata/server/coordinator/DefaultCore.java:266]: 在同步情况下如tcc或xa,是否应该在这succes且branch为empty时进行MetricsPublisher.postSessionDoneEvent?

Comment (slievrly) [reply server/src/main/java/io/seata/server/coordinator/DefaultCore.java]: done.

File: server/src/main/java/io/seata/server/coordinator/DefaultCore.java (hunk 1 above)
Comment (slievrly) [reply server/src/main/java/io/seata/server/coordinator/DefaultCore.java]: done

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by xingfudeshi — APPROVED** [review]
LGTM

**Pull request merged**